src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -96,8 +102,22 @@ public Task Start(IMessageSession session)
             }
         }
 
+        void SetupLongRunningWarning(Task taskToWatch, string message)
+        {
+            Task.Delay(LongRunningWarningTimeSpan)
+                .ContinueWith(t =>
+                {
+                    var taskIsFinished = taskToWatch.IsCompleted || taskToWatch.IsFaulted || taskToWatch.IsCanceled;
```

Comment:
No need to check for all three conditions, `IsCompleted` is true when any of the three final states have been reached. [Link](https://msdn.microsoft.com/en-us/library/system.threading.tasks.task.iscompleted(v=vs.110).aspx)

I guess I'm OK with this approach, but that does mean the task will stick around to run the check even when the most likely scenario is that we've completed a long time ago.